src/tests/test_resultset.py

```diff
@@ -109,177 +110,478 @@ def test_invalid_operation_error(result_set, fname, parameters):
+        "sqlite_sqlalchemy",
     ],
 )
-def test_lazy_loading_displaylimit_fetch_all(
-    result, config, displaylimit, expected_display
+def test_convert_to_dataframe_select(session, request):
+    session = request.getfixturevalue(session)
+
+    mock = Mock()
+    mock.displaylimit = 100
+    mock.autolimit = 100000
+
+    session.execute("CREATE TABLE a (x INT);")
+    session.execute("INSERT INTO a(x) VALUES (1),(2),(3),(4),(5);")
+    statement = "SELECT * FROM a"
+    results = session.execute(statement)
+
+    rs = ResultSet(results, mock, statement=statement, conn=session)
+    df = rs.DataFrame()
+
+    assert df.to_dict() == {"x": {0: 1, 1: 2, 2: 3, 3: 4, 4: 5}}
+
+
+@pytest.mark.parametrize(
+    "query",
+    [
+        "SELECT * FROM a",
+        "\nSELECT * FROM a",
+        "    SELECT * FROM a",
+        "FROM a",
+        "\nFROM a",
+        "    FROM a",
+    ],
+    ids=[
+        "select",
+        "select-with-newline",
+        "select-with-spaces",
+        "from",
+        "from-with-newline",
+        "from-with-spaces",
+    ],
+)
+@pytest.mark.parametrize(
+    "to_df_method, expected_value",
+    [
+        ("DataFrame", {"x": {0: 1, 1: 2, 2: 3, 3: 4, 4: 5}}),
+        ("PolarsDataFrame", {"x": [1, 2, 3, 4, 5]}),
+    ],
+)
+def test_convert_to_dataframe_using_native_duckdb(
+    ip_empty, query, to_df_method, expected_value
 ):
-    max_results_count = 3
-    config.autolimit = False
-    config.displaylimit = displaylimit
-    result_set = ResultSet(result, config)
-
-    # Initialize result_set without fetching results
-    assert len(result_set._results) == 0
-
-    # Fetch the min number of rows (based on configuration)
-    result_set.fetch_results()
-
-    html = result_set._repr_html_()
-    row_count = _get_number_of_rows_in_html_table(html)
-    expected_results = (
-        max_results_count
-        if expected_display + 1 >= max_results_count
-        else expected_display + 1
-    )
+    mock = Mock()
+    mock.displaylimit = 100
+    mock.autolimit = 100000
+
+    session = duckdb.connect()
+
+    session.execute("CREATE TABLE a (x INT);")
+    session.execute("INSERT INTO a(x) VALUES (1),(2),(3),(4),(5);")
+    results = session.execute(query)
+
+    rs = ResultSet(results, mock, statement=query, conn=DBAPIConnection(session))
+    # force fetching
+    list(rs)
+
+    df = getattr(rs, to_df_method)()
+
+    d = df.to_dict()
+
+    if to_df_method == "PolarsDataFrame":
+        d["x"] = list(d["x"])
+
+    assert d == expected_value
+
+
+def test_done_fetching_if_reached_autolimit(results):
+    mock = Mock()
+    mock.autolimit = 2
+    mock.displaylimit = 100
+
+    rs = ResultSet(results, mock, statement=None, conn=Mock())
+
+    assert rs.done_fetching() is True
+
+
+def test_done_fetching_if_reached_autolimit_2(results):
+    mock = Mock()
+    mock.autolimit = 4
+    mock.displaylimit = 100
+
+    rs = ResultSet(results, mock, statement=None, conn=Mock())
+    list(rs)
```

Comment:
I guess we use list and str to fetch the rest of the data. Can we add a comment?

Reply:
done